src/v/config/configuration.cc

```diff
@@ -1080,6 +1080,16 @@ configuration::configuration()
       false)
   , log_compaction_adjacent_merge_self_compaction_count(
       *this, "log_compaction_adjacent_merge_self_compaction_count")
+  , log_compaction_adjacent_merge_range_max(
+      *this,
+      "log_compaction_adjacent_merge_range_max",
+      "The maximum number of ranges of segments that can be processed in a "
+      "single round of adjacent segment compaction. If `null` (the default "
+      "value), no maximum is imposed on the number of ranges that can be "
+      "processed at once. A value below 1 effectively disables adjacent merge "
+      "compaction.",
+      {.needs_restart = needs_restart::no, .visibility = visibility::tunable},
+      std::nullopt)
   , log_compaction_adjacent_merge_segment_max(
```

Comment:
nit: IMO from the property names it's kind of hard to infer what these mean, and they feel quite easy to confuse with one another without reading the description. WDYT about something like:
- log_compaction_merge_max_segments_per_range
- log_compaction_merge_max_ranges

Then at least it's a bit easier to see the relationship/difference between them